Src/coffeescript/cql-execution/src/elm/aggregate.coffee

```diff
@@ -36,8 +57,10 @@ module.exports.Sum = class Sum extends AggregateExpression
   exec: (ctx) ->
     arg = @source.exec(ctx)
     if typeIsArray(arg)
-      filtered =  compact(arg)
-      if filtered.length == 0 then null else filtered.reduce (x,y) -> x+y
+      arg = compact(arg)
+      filtered =  quantitiesOrArg(arg)
```

Comment:
This pattern is seen a lot:
```
arg = compact(arg)
filtered =  quantitiesOrArg(arg)
```

In fact, `quantitiesOrArg` is _never_ called without a preceding `compact`.  As such, I'd suggest simplifying this so that the `quantitiesOrArg` does the compact.  Then you can always reduce this pattern to something more like:
```
filtered =  compactQuantitiesOrArg(arg)
```